pkg/query/table.go

```diff
@@ -242,13 +271,44 @@ func newTableBuilder(mem memory.Allocator) *tableBuilder {
 		builderCumulativeDiff:     rb.Field(schema.FieldIndices(TableFieldCumulativeDiff)[0]).(*builder.OptInt64Builder),
 		builderFlat:               rb.Field(schema.FieldIndices(TableFieldFlat)[0]).(*builder.OptInt64Builder),
 		builderFlatDiff:           rb.Field(schema.FieldIndices(TableFieldFlatDiff)[0]).(*builder.OptInt64Builder),
+		builderCallers:            rb.Field(schema.FieldIndices(TableFieldCallers)[0]).(*builder.ListBuilder),
+		builderCallees:            rb.Field(schema.FieldIndices(TableFieldCallees)[0]).(*builder.ListBuilder),
 	}
 
 	return tb
 }
 
+func (tb *tableBuilder) populateCallerAndCalleeData() {
+	for i := range tb.builderFunctionName.Len() {
+		if len(tb.callers) > i {
```

Comment:
I think we need a comment explaining these if statements, if I understand correctly, I think we need it to protect against the case where we didn't expand the callers to the row number (let's say the last row in the flamegraph table doesn't have callers or callees)?

Reply:
Done, added comments on these conditions.